lib/identity_cache/query_api.rb

```diff
@@ -161,13 +168,20 @@ def set_embedded_association(record, association_name, coder_or_array) #:nodoc:
       def get_embedded_association(record, association, options) #:nodoc:
         embedded_variable = record.public_send(options.fetch(:cached_accessor_name))
         if embedded_variable.respond_to?(:to_ary)
-          embedded_variable.map {|e| coder_from_record(e) }
+          embedded_variable.map {|e| coder_from_record_internal(e) }
         else
-          coder_from_record(embedded_variable)
+          coder_from_record_internal(embedded_variable)
         end
       end
 
       def coder_from_record(record) #:nodoc:
+        return unless record
+        ActiveSupport::Notifications.instrument('identity_cache.dehydration', class: record.class.name) do
```

Comment:
Similar to how I think we should use `deserialization` instead of `hydration`, I think this should be `serialization`